src/main/java/io/vertx/ext/mongo/impl/MongoClientImpl.java

```diff
@@ -905,8 +907,55 @@ private AggregatePublisher<JsonObject> doAggregate(final String collection, fina
     return aggregate;
   }
 
+
+
+  JsonArray deepEncodeKeyWhenUseObjectId(JsonArray arr) {
+    if(!useObjectId) return arr;
+
+    JsonArray newArr = new JsonArray(new ArrayList<>(arr.size()));
+
+    for (Object item : arr) {
+      if (item instanceof JsonArray) {
+        newArr.add(deepEncodeKeyWhenUseObjectId((JsonArray) item));
+      } else if (item instanceof JsonObject) {
+        newArr.add(deepEncodeKeyWhenUseObjectId((JsonObject) item));
+      } else {
+        newArr.add(item);
+      }
+      // we don't handle cases that value instanceof Map or List
```

Comment:
I do not convert map or list because I thought that even we can't assume how it's been created, the options are under control by developers and the could create a `good` JsonObject for this. Do not handle Map could keep the logic simple.
Anyway I could add logics to handle them.

Reply:
Yes please, add it